WordPress/src/main/java/org/wordpress/android/ui/mysite/SiteIconProgressSource.kt

```diff
@@ -0,0 +1,20 @@
+package org.wordpress.android.ui.mysite
+
+import androidx.lifecycle.distinctUntilChanged
+import androidx.lifecycle.map
+import kotlinx.coroutines.CoroutineScope
+import org.wordpress.android.ui.mysite.MySiteUiState.PartialState.ShowSiteIconProgressBar
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SiteIconProgressSource @Inject constructor(
```

Comment:
Added in 6ad3ddf. However it is really basic, because the tests are actually covered in SelectedSiteRepository, which gets us back to the question "What are sources and do we really need them". 🐔  & 🥚

Reply:
👋 @zwarm !

Thanks for adding the test, no matter if that is basic or not, we still have a test suite ready that we can now add on new tests when need be.

🙏

Reply:
@ParaskP7  
> What are sources and do we really need them 

Yeah, I was thinking over the weekend "Is MySiteSource just a marker interface to make it easier to request info at once?". Each source does different things. Some sources like 'QuickStartSource` - which was a part of QuickStartRepository do way more than something like `SelectedSiteSource`. We'll have an offline conversation about it. 

Thanks for the review. I'm going to wait for the new beta to be cut before merging. :)

Reply:
😃 

> Thanks for the review. I'm going to wait for the new beta to be cut before merging. :)

Great, yes, let's wait first and merge, 🟢 light from my side! 🚀